cpp/src/graph_utils.cuh

```diff
@@ -41,489 +40,554 @@ namespace cugraph
-    #if USE_CG
-      return __shfl_up_sync( mask, r, offset, bound );
-    #else
-      return __shfl_up( r, offset, bound );
-    #endif
-    #else
-      return 0.0f;
-    #endif
-}
-
-template<typename T>
-static __device__ __forceinline__ T shfl(T r, int lane, int bound = 32, int mask = DEFAULT_MASK)
-    {
-    #if __CUDA_ARCH__ >= 300
+	template<typename T>
+	static __device__  __forceinline__ T shfl_up(T r, int offset, int bound = 32, int mask =
+																									DEFAULT_MASK)
+																							{
+#if __CUDA_ARCH__ >= 300
+#if USE_CG
+		return __shfl_up_sync( mask, r, offset, bound );
+#else
+		return __shfl_up( r, offset, bound );
+#endif
+#else
+		return 0.0f;
+#endif
+	}
+
+	template<typename T>
+	static __device__  __forceinline__ T shfl(T r, int lane, int bound = 32, int mask = DEFAULT_MASK)
+																						{
+#if __CUDA_ARCH__ >= 300
 #if USE_CG
-        return __shfl_sync(mask, r, lane, bound );
+		return __shfl_sync(mask, r, lane, bound );
+#else
+		return __shfl(r, lane, bound );
+#endif
 #else
-        return __shfl(r, lane, bound );
+		return 0.0f;
 #endif
-    #else
-        return 0.0f;
-    #endif
-    }
-
-
-template<typename T>
-__inline__  __device__
-T parallel_prefix_sum(int n, int *ind,T *w) {
-    int i,j,mn;
-    T v,last; 
-    T sum=0.0;
-    bool valid;
-
-    //Parallel prefix sum (using __shfl)
-    mn =(((n+blockDim.x-1)/blockDim.x)*blockDim.x); //n in multiple of blockDim.x
-    for (i=threadIdx.x; i<mn; i+=blockDim.x) {
-        //All threads (especially the last one) must always participate 
-        //in the shfl instruction, otherwise their sum will be undefined.
-        //So, the loop stopping condition is based on multiple of n in loop increments,
-        //so that all threads enter into the loop and inside we make sure we do not 
-        //read out of bounds memory checking for the actual size n.
-
-        //check if the thread is valid
-        valid  = i<n;
-
-        //Notice that the last thread is used to propagate the prefix sum. 
-        //For all the threads, in the first iteration the last is 0, in the following 
-        //iterations it is the value at the last thread of the previous iterations. 
-        
-        //get the value of the last thread
-        last = shfl(sum, blockDim.x-1, blockDim.x);      
-
-        //if you are valid read the value from memory, otherwise set your value to 0
-        sum = (valid) ? w[ind[i]] : 0.0;
-
-        //do prefix sum (of size warpSize=blockDim.x =< 32)
-        for (j=1; j<blockDim.x; j*=2) {
-            v = shfl_up(sum, j, blockDim.x);
-            if (threadIdx.x >= j) sum+=v;
-        }
-        //shift by last
-        sum+=last;
-        //notice that no __threadfence or __syncthreads are needed in this implementation
-    }
-    //get the value of the last thread (to all threads)
-    last = shfl(sum, blockDim.x-1, blockDim.x);
-
-    return last;
-}
+	}
+
+	template<typename IdxType, typename ValType>
```

Comment:
It seems that the encoding or indentation has been overwritten in this file

Reply:
I think I re-formatted it. Does it need to be undone?

Reply:
It is best to avoid re-formatting because it is hard to see what changed and prevents auto-merging in case someone was editing this file too. 
If you have a quick way of undoing it, great. If not, I don't think you should spend a lot of time on that.

Reply:
Seems it is not straight forward to revert. We'll just keep the new version